feat: remove algosdk and update to the latest utils

neilcampbell opened the pull request.

tristanmenzel committed:
refactor!: Remove algosdk dependency

neilcampbell committed:
fix: updating to the latest utils and applying some fixes

neilcampbell committed:
chore: format struct types

neilcampbell committed:
chore: format struct types and fix return jsdocs

neilcampbell committed:
chore: add support for newGroup config

neilcampbell committed:
Merge pull request #241 from algorandfoundation/refactor/remove-algosdk

refactor!: Remove algosdk dependency

BREAKING CHANGE: removing algosdk

neilcampbell committed:
fix: remove dev dependency from non dev code

neilcampbell committed:
fix: remove dev dependency from non dev code

PatrickDinh committed:
* fix: add *FromTuple functions back

PatrickDinh committed:
* fix: add *FromTuple functions back

PatrickDinh committed:
fix: use original struct name in *FromTuple functions (#245)

PatrickDinh committed:
fix: use original struct name in *FromTuple functions (#245)

PatrickDinh commented on scripts/generate-clients.ts:
```diff
@@ -59,7 +59,4 @@ if (args.length === 0) {
-  colorConsole.error`${err}`
-  process.exit(-1)
-})
+void generateClients(appSpecPath)
```
Should we revert this back to have better error handling?

iglosiggio replied:
Done.

neilcampbell committed:
chore: update to the latest utils-ts

neilcampbell committed:
chore: update to the latest utils-ts

PatrickDinh committed:
fix: Changes during PR review (#246)

* fix: bug with storage key type

* chore: move bare call first (follow current main behaviour)

* build: update semantic release to v25

PatrickDinh committed:
fix: Changes during PR review (#246)

* fix: bug with storage key type

* chore: move bare call first (follow current main behaviour)

* build: update semantic release to v25

joe-p committed:
chore: update to alpha 29

joe-p committed:
chore: update to alpha 29

joe-p committed:
chore: add tslib

joe-p committed:
chore: add tslib

joe-p committed:
Merge pull request #248 from algorandfoundation/alpha-chore/utils_29

chore: update to utils alpha 29

PatrickDinh committed:
chore: remove 'types' from algokit-utils import

PatrickDinh committed:
fix: npm audit

PatrickDinh committed:
chore: ignore tar audit (#250)

iglosiggio committed:
chore: Bump algokit-utils

iglosiggio committed:
chore: npm audit fix

iglosiggio commented:
Rebased

iglosiggio committed:
chore: run `npm audit fix` and remove audit exceptions

iglosiggio committed:
chore: Re-add error reporting to scripts/generate-clients.ts

iglosiggio committed:
chore: Lint .ts files under the scripts folder

iglosiggio committed:
chore: Bump algokit-utils to the latest alpha

iglosiggio committed:
Merge v7.0.0-alpha.6 so semantic-release works again

iglosiggio committed:
chore: The previous commit was mine

I cannot go and sign the previous commit (since doing that will break the
already released version). This is just a note in the git history so anyone can
verify that I was the author of that commit (by signing this one and writing
this message).

The previous commit was created via `git commit-tree`. This porcelain command
does not respect `commit.gpgsign` (that makes sense, it is porcelain for a
reason).

iglosiggio committed:
chore: npm audit fix and exclusions for packages coming from npm itself
